systemtest/driver/flow.py

```diff
@@ -80,27 +82,60 @@ def check_max_agents(flow_dict, max_agents_dict):
     return success, agents
 
 
-def check_door_usage(flow_dict, num_agents):
-    success = True
-
-    # All pedestrian should leave through the main exit
-    exit_usage = max(flow_dict.values())  # maximum value
-    if exit_usage != num_agents:
-        success = False
-        logging.error(
-            "Number of agents exiting the simulation ({}) does not match number of agents in the simulation ({})".format(
-                exit_usage, num_agents
-            )
-        )
-
-    # All pedestirans should pass one of the other doors
-    other_door_usage = sum(flow_dict.values()) - exit_usage
-    if other_door_usage != num_agents:
-        success = False
-        logging.error(
-            "Number of agents using other doors ({}) does not match number of agents in the simulation ({})".format(
-                other_door_usage, num_agents
+def read_starting_times(events_file: Path):
+    tree = ET.parse(events_file)
+    root = tree.getroot()
+    starting_times = {}
+    for event in root.iter("event"):
+        if event.attrib["state"] == "open" or event.attrib["state"] == "reset":
+            starting_times.setdefault(int(event.attrib["id"]), []).append(
+                float(event.attrib["time"])
             )
-        )
 
-    return success
+    assert starting_times, "Could not read starting times from events file"
+    return starting_times
+
+
+def check_flow(
+    data_dict: dict,
+    max_agents_dict: dict,
+    starting_times_dict: dict,
+    outflow_dict: dict,
+):
```

Comment:
Please add a doctoring giving a high level explanation what this does, including the mitigation for many agents passing over the measurement line in a single frame.